src/classes/RD2_ScheduleService.cls

```diff
@@ -90,4 +263,33 @@ public without sharing class RD2_ScheduleService {
         return rd.npe03__Installment_Period__c == RD2_Constants.INSTALLMENT_PERIOD_FIRST_AND_FIFTEENTH;
     }
 
-}
+    /***
+    * @description Determines whether changes to recurring donation should trigger creation of new schedule(s)
+    * @param rd Trigger.new record
+    * @param oldRd Trigger.old record
+    * @return Boolean
+    */
+    public Boolean recurringDonationScheduleUpdateNeeded(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd) {
+
+        Boolean updateNeeded = false;
+
+        if (rd.Status__c == RD2_Constants.STATUS_CLOSED || rd.Status__c == RD2_Constants.STATUS_LAPSED) {
```

Comment:
so my take on your comment is that  I should also change getNewSchedules so that it does not create new schedules (leaving old schedule in place, but not create a new schedule).  Correct?

`    public List<RecurringDonationSchedule__c> getNewSchedules(npe03__Recurring_Donation__c rd) {

        List<RecurringDonationSchedule__c> newSchedules = new List<RecurringDonationSchedule__c>();

        if (rd.Status__c == RD2_Constants.STATUS_CLOSED || rd.Status__c == RD2_Constants.STATUS_LAPSED) {
            return newSchedules;
        }
`

Reply:
If the RD Status is changed from Active to Closed, then yes - apply EndDate to the current open schedules and not create a new schedule.
If the RD Status is changed from Active to Lapsed, then do nothing.

Reply:
shouldn't we also apply the end date if the change is from lapsed to closed?

Reply:
Yes. Good point. I left out that use case. I guess the first item should have been "If the RD Status is changed from Active/Lapsed to Closed, then apply EndDate to the current open schedules".